src/core/trulens/experimental/otel_tracing/core/instrument.py

```diff
@@ -81,3 +92,57 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return inner_decorator
+
+
+class App(core_app.App):
+    # For use as a context manager.
+    def __enter__(self):
+        logging.debug("Entering the OTEL app context.")
+
+        # Note: This is not the same as the record_id in the core app since the OTEL
+        # tracing is currently separate from the old records behavior
+        otel_record_id = str(uuid.uuid4())
+
+        tracer = trace.get_tracer_provider().get_tracer(TRULENS_SERVICE_NAME)
+
+        # Calling set_baggage does not actually add the baggage to the current context, but returns a new one
+        # To avoid issues with remembering to add/remove the baggage, we attach it to the runtime context.
```

Comment:
s/To/to

Reply:
I also don't really get this token business, can you enlighten me?

Reply:
from what I can tell - to set the baggage you need to:

1. call `set_baggage`, which returns a new, updated context
2. attach it to the current context with `context_api.attach` - this returns a token that can then be used to remove it.
3. Once the record is over, we want to remove it, and the way to do it via OTEL `context` is by telling it the baggage you want it to detach via `context.detach(token)`